### PR TITLE
[CPDLP-3105] TS7 - Add declaration seed data to separation

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3105

### Changes proposed in this pull request

* Updated `ValidTestDataGenerators::ApplicationsPopulater` to include declarations

### CODE REVIEW

**Comment by peteryates** [comment]
Something definitely feels a little odd about calling `FactoryBot` from `app/`.

Should these services be elsewhere?

**Comment by cwrw** [comment]
> Should these services be elsewhere?

Good point, but it felt like if we named it properly it would be ok for now as we wanted it to be a service that might be called from an endpoint later surfaced to providers on a sandbox environment

File: app/services/valid_test_data_generators/applications_populater.rb [hunk 1]
```diff
@@ -92,5 +93,17 @@ def reject_application(application)
       Applications::Reject.new(application:).reject
       application.reload
     end
+
+    def create_declarations(application)
+      %w[started retained-1 retained-2 completed].each do |declaration_type|
+        break if Faker::Boolean.boolean(true_ratio: 0.5)
+
+        FactoryBot.create(
+          :declaration,
```
[the remark this answers is not in the document]
Comment (cwrw) [reply app/services/valid_test_data_generators/applications_populater.rb:102]: ah applications are not always eligible it seems, could we make sure we pass the eligibility onwards to the declaration please?

**Comment by mooktakim** [comment]
> Something definitely feels a little odd about calling `FactoryBot` from `app/`.
> 
> Should these services be elsewhere?

if we think of seed data used for testing, it makes sense to use factory.

**Review by ethax-ross — COMMENTED** [review]
Looks good past Abeer's comments. I wasn't sure what this meant in the ticket though and if its already implemented elsewhere/still applies?

> Can we set limits for statements/applications as we already have providers set up on the separation environment for applications with declarations?

One for a future discussion, but I think we should maybe move these test data services out of `app/services` and into `lib` or somewhere else.

**Review by ethax-ross — APPROVED** [review]
Looks good to me

**Pull request merged**